controllers/callhomeconfig_controller.go

```diff
@@ -139,41 +143,40 @@ func (r *CallHomeConfigReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 			fmt.Sprintf("unexpected error getting service account: %s", err),
 			ctrl.Result{})
 	}
-	if len(sa.Secrets) < 1 {
-		return retryLater(r, log, &chc, "serviceaccount",
-			namespacedName.Name, "does not have a token")
-	}
-	tokenSecretName := sa.Secrets[0]
-	var token corev1.Secret
+	var kubeconfigSecret corev1.Secret
 	if err := r.Get(ctx, types.NamespacedName{
 		Namespace: req.Namespace,
-		Name:      tokenSecretName.Name,
-	}, &token); err != nil {
+		Name:      "cluster-autoscaler-mgmt-kubeconfig",
+	}, &kubeconfigSecret); err != nil {
 		if apierrors.IsNotFound(err) {
-			return retryLater(r, log, &chc, "token secret",
-				tokenSecretName.Name, "does not exist yet")
+			return retryLater(r, log, &chc, "secret",
+				namespacedName.Name, "does not exist yet")
 		}
 		return updateCallHomeConfigState(r, log, &chc, "error",
 			fmt.Sprintf("unexpected error getting service account: %s", err),
```

Comment:
That error message is now wrong (it talks about reading a service account)

Reply:
done